### ISSUE TITLE
Metrics for comparing targets and predictions

### ISSUE THREAD

**Issue by johannes-kk** [johannes-kk/cs205_final_project#51]
Create `metrics.cpp` and `metrics.hpp`, similar to `losses.*` but for comparing `targets` to `predictions`.

E.g. accuracy:

```
double accuracy_score(datavec targets, datavec predictions){
    // todo: verify that target and predict have same size
    double correct = 0;
    for (int i = 0; i < targets.size(); i++){
        if (targets[i] == predictions[i]){
            correct += 1;
        } 
    }
    return correct/targets.size();
}
```

**Comment by wfseaton** [comment]
Which metrics would we want? Here is sklearn's list for reference: https://scikit-learn.org/stable/modules/model_evaluation.html

**Comment by johannes-kk** [comment]
`accuracy` is all we need to begin with. `Precision`, `Recall`, `F1-score` and the likes would be nice to have, but nothing we need. We're focusing on binary classification for the MVP, and `accuracy` is plenty to make comparisons between methods and runs.
